windows.js

```diff
@@ -50,15 +54,70 @@ export async function install(platform, engine, version) {
 
   let toolchainPaths = (version === 'mswin') ? await setupMSWin() : await setupMingw(version)
 
-  common.setupPath([`${rubyPrefix}\\bin`, ...toolchainPaths])
-
   if (!inToolCache) {
     await downloadAndExtract(engine, version, url, base, rubyPrefix);
   }
 
+  const winMSYS2Type = common.setupPath([`${rubyPrefix}\\bin`, ...toolchainPaths])
+
+  // install msys2 tools for all Ruby versions, only install mingw or ucrt for Rubies >= 2.4
+
+  if (!['windows-2019', 'windows-2016'].includes(virtualEnv)) {
+    await installMSYS2Tools()
+  }
+
+  if ((( winMSYS2Type === 'ucrt64') || !['windows-2019', 'windows-2016'].includes(virtualEnv)) &&
+    (common.floatVersion(version) >= 2.4)) {
+    await installGCCTools(winMSYS2Type)
+  }
+
+  const ridk = `${rubyPrefix}\\bin\\ridk.cmd`
+  if (fs.existsSync(ridk)) {
+    await common.measure('Adding ridk env variables', async () => addRidkEnv(ridk))
+  }
+
   return rubyPrefix
 }
 
+// Actions windows-2022 image does not contain any mingw or ucrt build tools.  Install tools for it,
+// and also install ucrt tools on earlier versions, which have msys2 and mingw tools preinstalled.
+async function installGCCTools(type) {
+  const downloadPath = await common.measure(`Downloading ${type} build tools`, async () => {
+    let url = `https://github.com/MSP-Greg/setup-msys2-gcc/releases/download/msys2-gcc-pkgs/${type}.7z`
```

Comment:
@hsbt Are we OK to depend on a repository outside the ruby GitHub organization for setting up Ruby on Windows? Or should we move that repo to the ruby organization?
(Of course I trust @MSP-Greg and he is the best person to maintain it)

Reply:
I'm fine to merge as-is for now and we can figure that part later.

Reply:
Sorry, late response.

It's okay to merge this. I think it's better to transfer ruby org because we should support them by team members.

Reply:
That's fine with me.  No idea how to do so.  I'll see if I can find anything...

Reply:
Forgot one thing.  There is a folder in the repo, https://github.com/MSP-Greg/setup-msys2-gcc/tree/main/.github/actions/upload-binary.

It contains the js packaged code that uploads the asembled files to the release.  The code is very similar to the code I use for ruby-loco, so it's handled the code for Actions `mingw`, `ucrt`, and `mswin` Windows Rubies.

Let me see where that code is, as updates to the Actions packages normally warrant an update to it...

Reply:
@MSP-Greg Can you transfer them to me ( @hsbt )? After that, I will transfer to ruby organization. And I also create `actions` team and promote you and eregon to admin grant.

Reply:
@hsbt

I apologize for the delay.  I made many changes to both repos, and I wanted to update some docs re newer Rubies, ucrt, mswin, etc.  Some changes to mswin builds are not yet complete, but it's been too long already.

So, I'm ready to transfer them.  Is now a good time?

Thanks, Greg

Reply:
No problem. Can you transfer them to me? After that, I also transfer them to ruby org and I promote you and @eregon to their admin.

Reply:
👍 

I'm also done. Just available https://github.com/ruby/setup-ruby-pkgs and https://github.com/ruby/setup-msys2-gcc